spring-integration-core/src/test/java/org/springframework/integration/dsl/transformers/TransformerTests.java

```diff
@@ -287,6 +342,71 @@ public IdempotentReceiverInterceptor idempotentReceiverInterceptor() {
 			return idempotentReceiverInterceptor;
 		}
 
+		@Bean
+		public PollableChannel subFlowTestReplyChannel() {
+			return new QueueChannel();
+		}
+
+		@Bean
+		@Autowired
+		public IntegrationFlow syncSubFlowEnricher(SomeServiceGateway someServiceGateway) {
+			return f -> f
+					.enrich(e -> e.<TestPojo>requestPayload(p -> p.getPayload().getName())
+							.subFlow(sf -> sf
+									.<String>handle((p, h) -> someServiceGateway.deriveValueSync(p)))
+							.<String>headerFunction("foo", m -> m.getPayload())
+							.propertyFunction("name", m -> m.getPayload()))
+					.channel("subFlowTestReplyChannel");
+		}
+
+		@Bean
+		@Autowired
+		public IntegrationFlow asyncSubFlowEnricher(SomeServiceGateway someServiceGateway) {
```

Comment:
I initially had the requirement of enriching a message with information coming from a web resource and was thinking it was a good idea to make it an async call.  I realise now it doesn't makes sense since the whole enrich call is blocking - maybe i'm approaching the problem incorrectly.  Anyway, will still submit this change as it could be helpful to some.  I've applied the necessary changes to the test related to this.